enhanced-resolve/lib/Storage.d.ts

```diff
@@ -0,0 +1,34 @@
+/// <reference types="node" />
+import { CommonFileSystemMethod } from './common-types';
+
+declare class Storage {
+    duration: number;
+    running: {
+        [name: string]: ((this: null, ...args: any[]) => any)[]
+    };
+    data: {
+        [name: string]: any[]
+    };
+    levels: string[][];
+    count: number;
+    interval: NodeJS.Timer | null;
+    needTickCheck: boolean;
+    nextTick: number | null;
+    passive: boolean;
+
+    constructor(duration: number);
+
+    ensureTick(): void;
+
+    finished(name: string): void;
+
+    provide(name: string, provider: CommonFileSystemMethod, callback: (this: null, ...args: any[]) => any): any;
```

Comment:
Just use `(...args: any[]) => any`. `callback.apply(null, data)` is just an ES5 way of saying `callback(...data)`, as shown by some of the examples [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply).